### ISSUE TITLE
Reproducing AffectNet results.

### ISSUE THREAD

**Issue by sunggukcha** [kdhht2334/ELIM_FER#2]
Hello!

Thanks for sharing your works.

* How can I reproduce the affectnet results?
The script seems to require some preprocess.

* Can you share affectnet pretrained model?

Regards,
Sungguk Cha

**Comment by kdhht2334** [comment]
Hi,

First of all, I upload `elim_category.py` and update corresponding details. Please check.

**Note**
You _must_ add estimated age scores to your `train/val.script`. And you may choose two options.
1) Use pretrained age estimation model to complete your `train/val.script` (link: https://github.com/yu4u/age-estimation-pytorch)
2) Borrow both `age_script/training_age.csv` and `age_script/validation_age.csv` to complete your `train/val.script`
- Other metadata can be obtained from AffectNet official website :)

Unfortunately, I cannot share pretrained model of AffectNet..I cannot find the weights for now.
But I can tell that you can easily get following bench-marking Table using `elim_category.py`!

| Method | Acc. |
| --- | --- |
|Baseline [1]|0.58|
|VGG-Face [2]|0.60|
|HO-Conv [3]|0.59|
|ELIM-Age (R18)|0.611|
|Face2Exp [4]|0.64|

**References**
[1] A. Mollahosseini et al., Affectnet: A database for facial expression, valence, and arousal computing in the wild, TAC 2017.
[2] D. Kollias et al., Generating faces for affect analysis, ArXiv 2018.
[3] J. Kossaifi et al., Factorized higher-order cnns with an application to spatio-temporal emotion estimation, CVPR 2020.
[4] D. Zeng et al., Face2Exp: combating data biases for facial expression recognition, CVPR 2022.

**Comment by sunggukcha** [comment]
Thanks for your reply.
My interest was Valence-Arousal prediction in AffectNet dataset.
In this case, can I get helped?

**Comment by kdhht2334** [comment]
AffectNet was mainly used for **classification purposes**. However, you can also use it for valence-arousal regression purposes as follows.

For convenience, I update the `elim_age.py` file. Please refer to this file for AffectNet.

Also i'm inspired following link:
* https://github.com/behzadhsni/BReG-NeXt

**Comment by sunggukcha** [comment]
Sincerely thanks for your kind reply. 

It is rather far from the question.
"why do you prefer `Aff-wild2` or `AFEW-VA` dataset than `AffectNet (valence-arousal)`?"
In my perspective of view, AffectNet has (1) more number of frames and (2) more number of identities, though it does not have temporal information. 
I want your opinion on the values of the `Aff-Wild` and `AFEW-VA`.

**Comment by kdhht2334** [comment]
1) `AFEW-VA` is suitable for fast check of model performance in exchange for a small number of frames. On the other hand, if you pre-process `Aff-wild2`'s video clips, you can get about 1.4M frames. It is the most large-scale VA databse so far. Since these two datasets are from emotion recognition challenge (or variants), i think they are classified as wild DB.
   - `AffectNet` is also a well-refined wild DB, but the domain gap between train and test datasets is large. I think AffectNet's performance is somewhat saturated. The reason is that the SOTA 7-class accuracy (%) of this AffectNet is less than 70.

2) Many IDs can be obtained from the AffectNet dataset, but the problem is that **only 1 (or 2-3) samples are given for each ID**. This fact becomes a major obstacle to the implementation of ELIM, which organizes datasets by domain. Therefore, I focused on AFEW-VA and Aff-wild2 given the raw data as video clips.
  - cf) So, we had no choice but to experiment with AffectNet using the `age group` as the domain.

**Comment by sunggukcha** [comment]
You helped much.
It was really nice for me to talk with you, @kdhht2334 .

Hope you happy.
Regards,
Sungguk Cha
